src/ImportDefinitionsBundle/Resources/config/services.yml

```diff
@@ -186,5 +196,7 @@ services:
     tags:
       - { name: import_definition.setter, type: object_type, form-type: ImportDefinitionsBundle\Form\Type\NoConfigurationType }
 
-
-
+  import_definition.setter.add_to_array:
+    class: ImportDefinitionsBundle\Setter\ObjectType
```

Comment:
Setter is not right configured here. Your class is named AddToArray and you configure ObjectType here.

Anyway: I don't see any sense why you are adding a new setter? Setter should actually apply the value the object/fieldcollection/localizedfield and so on. What you are doing is more like a InterpreterWrapper, right?

Reply:
What's your proposition to handle objects and multihref fields?

Reply:
chained interpreter would be a solution.